geary-papermc-features/src/main/kotlin/com/mineinabyss/geary/papermc/features/common/actions/BoneMealRadiusAction.kt

```diff
@@ -0,0 +1,48 @@
+package com.mineinabyss.geary.papermc.features.common.actions
+
+import com.mineinabyss.geary.actions.Action
+import com.mineinabyss.geary.actions.ActionGroupContext
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import net.minecraft.tags.BlockTags
+import org.bukkit.Material
+import org.bukkit.Tag
+import org.bukkit.block.Block
+import org.bukkit.block.BlockFace
+import org.bukkit.entity.Player
+
+
+@Serializable
+@SerialName("geary:bonemeal_radius")
+class BoneMealRadiusAction(
+    val radius: Int,
+    val AllowedTags: Set<String>,
+    val excludedBlocks: Set<String>,
+) : Action {
+    override fun ActionGroupContext.execute() {
+        val player = entity?.get<Player>() ?: return println("No player entity found for BoneMealRadiusAction")
+        val center = player.location.block
+        val world = player.world
+
+        val allowedBlockTags = mutableSetOf<Tag<Material>>()
+        for (tag in AllowedTags) {
+            when (tag.lowercase()) {
+                "saplings" -> allowedBlockTags.add(Tag.SAPLINGS)
+                "crops" -> allowedBlockTags.add(Tag.CROPS)
+            }
+        }
```

Comment:
not too sure what this aims to do, if it only aims to support saplings and crops you can do so without a large list like this as input. just make two booleans or something